From 4fb6be44072d48504256ff663128bc062c3c36b4 Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Fri, 12 Sep 2014 14:26:47 +0900
Subject: [PATCH 11/22] vsp-renderer: open subdev via v4l2_subdev.

Avoid a possibility to open the same subdev nodes multiple times.
---
 src/vsp-renderer.c |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/vsp-renderer.c b/src/vsp-renderer.c
index 869d412..8041e39 100644
--- a/src/vsp-renderer.c
+++ b/src/vsp-renderer.c
@@ -301,14 +301,13 @@ vsp_init(struct media_device *media)
 			goto error;
 		}
 
-		devname = media_entity_get_devname(entity);
-		weston_log("input '%s' is associated with '%s'\n", buf, devname);
-		vsp->input_pads[i].fd = open(devname, O_RDWR);
-		if (vsp->input_pads[i].fd < 0) {
-			weston_log("error... can't open '%s'.\n", devname);
+		if (v4l2_subdev_open(entity)) {
+			weston_log("subdev '%s' open failed\n.", buf);
 			goto error;
 		}
-		vsp_check_capabiility(vsp->input_pads[i].fd, devname);
+
+		vsp->input_pads[i].fd = entity->fd;
+		vsp_check_capabiility(vsp->input_pads[i].fd, media_entity_get_devname(entity));
 	}
 
 	/* Initialize output */
-- 
1.7.9.5

